ocean_provider/util_url.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import ipaddress
+import dns.resolver
+from ocean_provider.utils.basics import get_config
+
+from urllib.parse import urlparse
+
+logger = logging.getLogger(__name__)
+
+
+def is_safe_url(url):
+    if not is_safe_schema(url):
+        return False
+    result = urlparse(url)
+    return is_safe_domain(result.netloc)
+
+
+def is_safe_schema(url):
+    try:
+        result = urlparse(url)
+        return all([result.scheme, result.netloc, result.path])
+    except:
+        return False
+
+
+def is_safe_domain(domain):
+    DNS_RESOLVER = dns.resolver.Resolver()
+    try:
+        ip_v4_records = DNS_RESOLVER.resolve(domain, "A", search=True)
+    except Exception as e:
+        ip_v4_records = None
+        logger.info(
+            "[i] Cannot get A record for domain '%s': %s\n" % (domain, e))
+    try:
+        ip_v6_records = DNS_RESOLVER.resolve(domain, "AAAA", search=True)
+    except Exception as e:
+        ip_v6_records = None
+        logger.info(
+            "[i] Cannot get AAAA record for domain '%s': %s\n" % (domain, e))
+
+    if verify_dns_records(domain, ip_v4_records, "A") or verify_dns_records(domain, ip_v6_records, "AAAA"):
+        return False
+    return True
+
+
+def verify_dns_records(domain, records, type):
```

Comment:
can you rename the "type" parameter into something more explicit? I also think "type" is a reserved keyword and might get us into trouble with Python

Reply:
type is the record type.  we need to check for both "A"  (ipv4) and "AAAA" (ipv6)

Reply:
then how about record_type?